src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SslOverTdsStream.NetCoreApp.cs

```diff
@@ -0,0 +1,305 @@
+            }
+
+            using (SNIEventScope.Create("<sc.SNI.SslOverTdsStream.Read |SNI|INFO|SCOPE> reading encapsulated bytes"))
+            {
+                if (_packetBytes > 0)
+                {
+                    // there are queued bytes from a previous packet available
+                    // work out how many of the remaining bytes we can consume
+                    int wantedCount = Math.Min(buffer.Length, _packetBytes);
+                    int readCount = _stream.Read(buffer.Slice(0, wantedCount));
+                    if (readCount == 0)
+                    {
+                        // 0 means the connection was closed, tell the caller
+                        return 0;
+                    }
+                    _packetBytes -= readCount;
+                    return readCount;
+                }
+                else
+                {
+                    Span<byte> headerBytes = stackalloc byte[TdsEnums.HEADER_LEN];
+
+                    // fetch the packet header to determine how long the packet is
+                    int headerBytesRead = 0;
+                    do
+                    {
+                        int headerBytesReadIteration = _stream.Read(headerBytes.Slice(headerBytesRead, TdsEnums.HEADER_LEN - headerBytesRead));
+                        if (headerBytesReadIteration == 0)
+                        {
+                            // 0 means the connection was closed, tell the caller
+                            return 0;
+                        }
+                        headerBytesRead += headerBytesReadIteration;
+                    } while (headerBytesRead < TdsEnums.HEADER_LEN);
+
+                    // read the packet data size from the header and store it in case it is needed for a subsequent call
+                    _packetBytes = ((headerBytes[TdsEnums.HEADER_LEN_FIELD_OFFSET] << 8) | headerBytes[TdsEnums.HEADER_LEN_FIELD_OFFSET + 1]) - TdsEnums.HEADER_LEN;
+
+                    // read as much from the packet as the caller can accept
+                    int packetBytesRead = _stream.Read(buffer.Slice(0, Math.Min(buffer.Length, _packetBytes)));
+                    _packetBytes -= packetBytesRead;
+                    return packetBytesRead;
+                }
+            }
+        }
+
+        public override async ValueTask<int> ReadAsync(Memory<byte> buffer, CancellationToken cancellationToken = default)
+        {
+            if (!_encapsulate)
+            {
+                int read;
+                {
+                    ValueTask<int> readValueTask = _stream.ReadAsync(buffer, cancellationToken);
+                    if (readValueTask.IsCompletedSuccessfully)
+                    {
+                        read = readValueTask.Result;
+                    }
+                    else
+                    {
+                        read = await readValueTask.ConfigureAwait(false);
+                    }
+                }
+                return read;
+            }
```

Comment:
No, because it would fire for every single packet sent or received which would certainly have performance implications and the underlying SslStream will have it's own logging. This class is only interesting when it is encapsulating, otherwise it should do as little as possible.